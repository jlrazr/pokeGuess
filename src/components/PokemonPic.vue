<template>
  <div class="img-container">
    <img class="hidden" :src="imgSrc" alt="Guess the hidden Pokémon" />
    <img
      v-if="reveal"
      class="fade-in"
      :src="imgSrc"
      alt="Guess the hidden Pokémon"
    />
  </div>
</template>

<script>
export default {
  props: {
    pokeId: {
      type: Number,
      required: true,
    },
    reveal: {
      type: Boolean,
      required: true,
      default: false,
    },
  },

  computed: {
    imgSrc() {
      const img = `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokeId}.svg`;
      return img;
    },
  },
};
</script>

<style lang="scss" scoped>
img {
  position: absolute;
  height: 110px;

  @media (min-width: 766px) {
    height: 200px;
  }
}

.img-container {
  position: relative;
  margin: 0 auto;
  display: flex;
  justify-content: center;
  height: 120px;

  @media (min-width: 766px) {
    height: 200px;
  }
}

.hidden {
  filter: brightness(0);
  -moz-user-drag: none;
  -webkit-user-drag: none;
}

.remove-hidden {
  opacity: 1;
}
</style>
