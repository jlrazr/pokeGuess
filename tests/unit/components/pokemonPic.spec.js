import { shallowMount } from "@vue/test-utils";
import PokemonPic from "@/components/PokemonPic";

describe("PokemonPic Component", () => {
  test("must match the snapshot", () => {
    const wrapper = shallowMount(PokemonPic, {
      props: {
        pokeId: 150,
        reveal: false,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  test("must show a hidden/black image and pokemon with ID 150", () => {
    const wrapper = shallowMount(PokemonPic, {
      props: {
        pokeId: 150,
        reveal: false,
      },
    });

    const [image1, image2] = wrapper.findAll("img");
    expect(image1.exists()).toBeTruthy();
    expect(image1.classes("hidden")).toBe(true);
    expect(image2).toBe(undefined);
    expect(image1.attributes().src).toBe(
      "https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/150.svg"
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("must show the Pokemon if revealPokemon = true", () => {
    const wrapper = shallowMount(PokemonPic, {
      props: {
        pokeId: 150,
        reveal: true,
      },
    });

    console.log(wrapper.html());
    const [image1, image2] = wrapper.findAll("img");
    expect(image1.exists()).toBeTruthy();
    expect(image2.classes("fade-in")).toBe(true);
  });
});
