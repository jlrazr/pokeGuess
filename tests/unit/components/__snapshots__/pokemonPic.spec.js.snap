// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPic Component must match the snapshot 1`] = `
<div class="img-container"><img class="hidden" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/150.svg" alt="Guess the hidden Pokémon">
  <!--v-if-->
</div>
`;

exports[`PokemonPic Component must show a hidden/black image and pokemon with ID 150 1`] = `
<div class="img-container"><img class="hidden" src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/150.svg" alt="Guess the hidden Pokémon">
  <!--v-if-->
</div>
`;
